packages/daemon/src/services/index.ts

```diff
@@ -601,6 +733,34 @@ export const voidTx = async (
   // This ensures the UTXOs can be used in new transactions after the void
   await clearTxProposalForVoidedTx(mysql, txInputs);
 
+  /**
+   * Delete ALL tokens created by this voided transaction.
+   *
+   * This handles all three token creation scenarios:
+   *
+   * 1. Pure CREATE_TOKEN_TX (no nano):
+   *    - Deletes the single CREATE_TOKEN_TX token (token_id = tx_id)
+   *
+   * 2. Pure nano contract:
+   *    - Deletes all tokens created by nano syscalls (token_id ≠ tx_id)
+   *
+   * 3. Hybrid (CREATE_TOKEN_TX + nano):
+   *    - Deletes BOTH the CREATE_TOKEN_TX token AND all nano-created tokens
+   *
+   * Note: This is INDEPENDENT of nano execution state (nc_execution).
+   * Even if nc_execution = 'success', we delete all tokens because the
+   * ENTIRE transaction is being voided.
+   *
+   * See handleVertexAccepted for nano execution state change logic, which
+   * ONLY deletes nano-created tokens when nc_execution becomes non-SUCCESS.
+   */
+  const tokensCreated = await getTokensCreatedByTx(mysql, hash);
+  if (tokensCreated.length > 0) {
+    logger.debug(`Voiding transaction ${hash} created ${tokensCreated.length} token(s), deleting them`);
+    await deleteTokens(mysql, tokensCreated);
+    await deleteTokenCreationMappings(mysql, tokensCreated);
```

Comment:
Just checking, the `token_creation` table will delete the entry when the `token_id` entry on the token table is deleted (because of the onCascade).
So just deleting the token should delete the mappings as well, correct?
Maybe we should check how many entries the `deleteTokenCreationMappings` is affecting to see if the mappings are already deleted.